openvasreporting/libs/export.py

```diff
@@ -45,12 +51,25 @@ def _get_collections(vuln_info):
     vuln_levels = Counter()
     vuln_host_by_level = Counter()
     vuln_by_family = Counter()
+    # collect host names
+    vuln_hostcount_by_level =[[] for _ in range(5)]
+    level_choices = {'critical': 0, 'high': 1, 'medium': 2, 'low': 3, 'none': 4}
 
     for i, vuln in enumerate(vuln_info, 1):
         vuln_levels[vuln.level.lower()] += 1
-        vuln_host_by_level[vuln.level.lower()] += len(vuln.hosts)
+        # add host names to list so we count unquie hosts per level
+        level_index = level_choices.get(vuln.level.lower())
+
+        for i, (host, port) in enumerate(vuln.hosts, 1):    
+            if host.ip not in vuln_hostcount_by_level[level_index]:
+                vuln_hostcount_by_level[level_index].append(host.ip)       
```

Comment:
Couldn't you use 

    vuln_hostcount_by_level[vuln.level.lower()].append(host.ip)

Reply:
No  vuln_hostcount_by_level is an array of lists so we need the index to be an int and vuln.level is a string, hence the use of the choice clause above, to get an appropriate int.